twine/__main__.py

```diff
@@ -24,9 +25,19 @@
 
 def main() -> Any:
     try:
-        return cli.dispatch(sys.argv[1:])
-    except (exceptions.TwineException, requests.HTTPError) as exc:
-        return _format_error(f"{exc.__class__.__name__}: {exc.args[0]}")
+        result = cli.dispatch(sys.argv[1:])
+    except requests.HTTPError as exc:
+        status_code = exc.response.status_code
+        status_phrase = http.HTTPStatus(status_code).phrase
+        result = (
+            f"{exc.__class__.__name__} from {exc.response.url}: "
+            f"{status_code} {status_phrase}\n"
+            f"{exc.response.reason}"
+        )
+    except exceptions.TwineException as exc:
+        result = f"{exc.__class__.__name__}: {exc.args[0]}"
+
+    return _format_error(result) if isinstance(result, str) else result
```

Comment:
Refactored in 3fd259ae162c966c7a7c49dfb2ce880301773dd5 to one `return` and one `_format_error` to ensure that any current and future string results are treated as an error, which is how they're interpreted by `sys.exit`.